script.js

```diff
@@ -1,4 +1,146 @@
+// Global variables
+let currentPlayer = 1;
+let numOfPlayers = 3;
+var numOfDice = 2;
+
+// declare 2 game modes so that the submit button will call main function 2 different ways
+var currGameMode = "roll dice"; // second game mode will be "choose order". Use if statements for this!
+
+var dice1 = "";
+var dice2 = "";
+
+var playerNumbers = [];
+var winnersArr = [];
+
+// Helper Functions
+var generateRandomNumber = function () {
+  var dicevalue = Math.floor(Math.random() * 6) + 1;
+  return dicevalue.toString(); // to combine the values via two options. Number data types don't work like that
+};
+
+var areDicesSame = function () {
+  if (dice1 == dice2) {
+    return true;
+  }
+  return false;
+};
+
+// reset game
+var resetGame = function () {
+  currGameMode = "roll dice";
+  currentPlayer = 1;
+  playerNumbers = [];
+  winnersArr = [];
+};
+
+// Remember to add a reset function here to reset the game after it ends
+
+// main function
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  if (currentPlayer > numOfPlayers) {
+    var highScore = Math.max(...playerNumbers);
+    console.log(highScore);
+
+    for (let i = 0; i < playerNumbers.length; i += 1) {
+      if (playerNumbers[i] == highScore) {
+        winnersArr.push("" + (i + 1));
+      }
+    }
+
+    if (winnersArr.length == 1) {
+      myOutputValue = `The winner is Player ${winnersArr}. Press submit to play again!`;
+    } else if (winnersArr.length < numOfPlayers) {
+      myOutputValue = `The winners are Player ${winnersArr[0]} and Player ${winnersArr[1]}. Press submit to play again!`;
+    } else if (winnersArr.length == numOfPlayers) {
+      myOutputValue = `Everyone drew. What were the chances of that happening! Press submit to play again!`;
+    }
+
+    resetGame();
+
+    return myOutputValue;
+  }
+
+  // if current game mode is roll dice
+  if (currGameMode == "roll dice") {
+    if (input != "") {
+      return `Please roll dice by clicking Submit`;
+    }
+    dice1 = generateRandomNumber();
+    dice2 = generateRandomNumber();
+    console.log("dice 1: " + dice1);
+    console.log("dice 2: " + dice2);
+
+    // check if dices are the same. If so, add both of them together
+    if (areDicesSame()) {
+      if (currentPlayer < numOfPlayers) {
+        playerNumbers.push(dice1 + dice2);
+        myOutputValue = `Your number is ${
+          playerNumbers[currentPlayer - 1]
+        }. It is now Player ${currentPlayer + 1}'s turn`;
+        currentPlayer += 1;
+      } else if ((currentPlayer = numOfPlayers)) {
```

Comment:
There is a logic error here due to your syntax. it should be a double equal sign